hal/src/lib.rs

```diff
@@ -97,29 +80,23 @@ pub mod time;
 pub mod timer_params;
 pub mod timer_traits;
 
-#[cfg(all(feature = "unproven", feature = "dma"))]
+#[cfg(feature = "dma")]
 pub mod dmac;
 
-#[cfg(all(feature = "usb", feature = "samd11"))]
-compile_error!("'usb' is enabled, but USB isn't supported on SAMD11");
-
-#[cfg(all(
-    feature = "usb",
-    not(any(feature = "samd21", feature = "min-samd51g", feature = "library"))
-))]
-compile_error!("The 'usb' feature is enabled, but not a chip with USB support");
+#[cfg(all(feature = "usb", feature = "device", not(feature = "usbp")))]
+compile_error!("The 'usb' feature is enabled, but this chip does not support USB");
```

Comment:
we should have one of these for sdmmc and other stuff too maybe?

Reply:
Yes, absolutely. I'll see what other instances make sense.